src/Data/Morpheus/Server.hs

```diff
@@ -58,44 +60,54 @@ import           Data.Morpheus.Types.Internal.AST
                                                 ( Value )
 
 handleSubscription
-  :: (Eq (StreamChannel e), GQLChannel e)
-  => GQLClient IO e
-  -> GQLState IO e
+  :: (Eq (StreamChannel e), GQLChannel e, MonadIO m)
+  => GQLClient m e
+  -> GQLState m e
   -> Text
-  -> ResponseStream e IO Value
-  -> IO ()
+  -> ResponseStream e m Value
+  -> m ()
 handleSubscription GQLClient { clientConnection, clientID } state sessionId stream
   = do
     response <- runResultT stream
     case response of
       Success { events } -> mapM_ execute events
-      Failure errors     -> sendTextData
+      Failure errors     -> liftIO $ sendTextData
         clientConnection
         (toApolloResponse sessionId $ Errors errors)
  where
   execute (Publish   pub) = publishUpdates state pub
   execute (Subscribe sub) = addClientSubscription clientID sub sessionId state
 
 -- | Wai WebSocket Server App for GraphQL subscriptions
-gqlSocketApp
-  :: RootResCon IO e que mut sub
-  => GQLRootResolver IO e que mut sub
-  -> GQLState IO e
+gqlSocketApp'
```

Comment:
maybe we can come up with better name, and export it

Reply:
Addressed here: https://github.com/morpheusgraphql/morpheus-graphql/pull/324/commits/f4cff6dbd6896ab81daa6f5bd04556d0e03cc6ce